app/services/api/beta/search_result_statistics_service.rb

```diff
@@ -0,0 +1,91 @@
+module Api
+  module Beta
+    class SearchResultStatisticsService
+      def initialize(goods_nomenclature_hits)
+        @goods_nomenclature_hits = goods_nomenclature_hits
+        @chapter_statistics = Hashie::TariffMash.new
+        @heading_statistics = Hashie::TariffMash.new
+      end
+
+      def call
+        search_result_statistics = []
+
+        search_result_statistics << generate_chapter_statistics
+        search_result_statistics << generate_heading_statistics
+
+        search_result_statistics
+      end
+
+      private
+
+      attr_reader :goods_nomenclature_hits
+
+      def generate_chapter_statistics
+        goods_nomenclature_hits.each do |goods_nomenclature_hit|
```

Comment:
I am happy with shortening a var  when something it is only inside a block, and the context is clear:

`goods_nomenclature_hits.each do |gn_hit|`

I guess this is personal preference too :)